python/pyspark/sql/pandas/serializers.py

```diff
@@ -298,26 +299,39 @@ class ArrowStreamPandasUDFSerializer(ArrowStreamPandasSerializer):
     Serializer used by Python worker to evaluate Pandas UDFs
     """
 
-    def __init__(self, timezone, safecheck, assign_cols_by_name, df_for_struct=False):
+    def __init__(
+        self,
+        timezone,
+        safecheck,
+        assign_cols_by_name,
+        df_for_struct=False,
+        eval_type=PythonEvalType.SQL_SCALAR_PANDAS_UDF,
```

Comment:
It shouldn't take `eval_type` here as the same as `df_for_struct` that is calculated from the `eval_type` in `worker.py`.

Reply:
Good point, adjusted.

Reply:
Docstrings on how `df_for_struct` and `struct_in_pandas` is set according to eval_types, as well as user-facing documentation on struct type behaviors would be helpful. I may work on it as a follow-up.